install.sh

```diff
@@ -0,0 +1,152 @@
+#! /usr/bin/env bash
+
+set -euo pipefail
+
+cat <<EOF
+     _           _   _   _      
+ ___| |__  _   _| |_| |_| | ___ 
+/ __| '_ \\| | | | __| __| |/ _ \\
+\__ \\ | | | |_| | |_| |_| |  __/
+|___/_| |_|\\__,_|\\__|\\__|_|\\___|
+                                
+https://www.shuttle.rs
+https://github.com/shuttle-hq/shuttle
+
+Please file an issue if you encounter any problems!
+===================================================
+EOF
+
+if ! command -v curl &>/dev/null; then
+  echo "curl not installed. Please install curl."
+  exit
+fi
+
+REPO_URL="https://github.com/shuttle-hq/shuttle"
+LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' "$REPO_URL/releases/latest")
+LATEST_VERSION="${LATEST_RELEASE#*\"tag_name\":\"}"
+LATEST_VERSION="${LATEST_VERSION%%\"*}"
+
+_install_linux() {
+  echo "Detected Linux!"
+  echo "Checking distro..."
+  if (uname -a | grep -qi "Microsoft"); then
+    OS="ubuntuwsl"
+  elif ! command -v lsb_release &>/dev/null; then
+    echo "lsb_release could not be found. Falling back to /etc/os-release"
+    OS="$(grep -Po '(?<=^ID=).*$' /etc/os-release | tr '[:upper:]' '[:lower:]')" 2>/dev/null
+  else
+    OS=$(lsb_release -i | awk '{ print $3 }' | tr '[:upper:]' '[:lower:]')
+  fi
+  case "$OS" in
+  "arch" | "manjarolinux" | "endeavouros")
+    _install_arch_linux
+    ;;
+  "ubuntu" | "ubuntuwsl" | "debian" | "linuxmint" | "parrot" | "kali" | "elementary" | "pop")
+    # TODO: distribute .deb packages via `cargo-deb` and install them here
+    _install_unsupported
+    ;;
+  *)
+    _install_unsupported
+    ;;
+  esac
+}
+
+_install_arch_linux() {
+  echo "Arch Linux detected!"
+  if command -v pacman &>/dev/null; then
+    echo "Installing with pacman"
+    sudo pacman -S cargo-shuttle
+  else
+    echo "Pacman not found"
+    exit 1
+  fi
+}
+
+# TODO: package cargo-shuttle for Homebrew
+_install_mac() {
+  _install_unsupported
+}
+
+_install_binary() {
+  echo "Installing pre-built binary..."
+  case "$OSTYPE" in
+  linux*) target="x86_64-unknown-linux-musl" ;;
+  darwin*) target="x86_64-apple-darwin" ;;
```

Comment:
There is an `aarch64` linux binary. Just me or is there no binary for M1/M2? I thought we built binaries cause they were struggling.

Reply:
We didn't implement a binary build for M1/M2 yet, because they can use the x86_64 binary, and it was a bit tricky to do at the time with CircleCI. However, we now use an M1 machine to build the x86_64 binary, so adding an m1 binary should be quite simple, essentially just copying the current mac build, removing the rosetta install step and changing the target.

Reply:
Or do both in one go to save on that juicy mac CI cost :+1: